src/pybind/cudamatrix/cu_device_pybind.cc

```diff
@@ -36,6 +36,16 @@ void pybind_cu_device(py::module& m) {
         },
         py::arg("device_id"));
 
+  m.def("SelectGpuId",
+        [](std::string use_gpu) {
```

Comment:
`const std::string& use_gpu`

The reason to use a specified GPU rather than an automatically selected
GPU is that we want to share data between PyTorch GPU tensor and CuSubMatrix/CuSubVector.

Reply:
Your commit may be useful in some cases. @danpovey

Reply:
Is there a plan to submit pytorch traning task distributed?

Reply:
We indeed have one internally, which can use either horovod or DDP (Distributed Data Parallel).
I do not know whether @danpovey is interested in distributed training with PyTorch.

Reply:
Cool!